### PR TITLE
Add in a kraken follows endpoint implementation

### PR DESCRIPTION
**Pull request by albert-wang**

### CODE REVIEW

File: follow.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package twitch
 
 import (
-	"fmt"
+    "fmt"
```
Comment (cwc) [inline follow.go:4]: Needs a go fmt?

**Review by cwc — APPROVED** [review]
Approved with go fmt 👍

File: twitch.go [hunk 2]
```diff
@@ -26,10 +27,11 @@ type RequestOptions struct {
 	Extra     *url.Values
 }
 
-func NewTwitchClient(clientID string) TwitchClient {
+func NewTwitchClient(clientID string, token string) TwitchClient {
```
Comment (camr) [inline twitch.go:30]: This is an API change. I forgot Go doesn't have optional arguments 🙄 

What are your thoughts on a new `NewTwitchClientWithOptions` that takes some amount of optional arguments including an `http.Client` and `bearer token`?

**Review by camr — CHANGES_REQUESTED** [review]

File: twitch.go (hunk 2 above)
Comment (cwc) [reply twitch.go:30]: OAuth tokens are required to do anything with the API now right? So all existing projects using the lib are broken and need updating anyway. The backwards-compatible func would have no use.

**Pull request closed without merging**